main.js

```diff
@@ -1,11 +1,188 @@
-var app = new Vue({
-    el: "#app",
-    data: {
-        nums: []
+
+
+
+Vue.component('Transactions', {
+    template: `#transactions-template`,
+    props:
+    {
+        finalTransaction: Array,
+        editTransactionId: Number,
+        
     },
-    methods: {
-        addNum() {
-            this.nums.push(Math.round(Math.random() * 10));
+    data() {
+        return {
+            id: null,
+            name: null,
+            description: null,
+            type: null,
+            amount: null,
+        }
+    },
+    watch: {
+        editTransactionId: function (_editTransactionId)
+        {
+            if (_editTransactionId)
+            {
+                const transaction = this.finalTransaction.find(t => t.id === _editTransactionId);
+                this.name = transaction.name;
+                this.description = transaction.description;
+                this.type = transaction.type;
+                this.amount = transaction.amount;
+                return;
+            }
         }
     },
-});
\ No newline at end of file
+    methods: {
+
+        editTransactionDisplay (){
+            this.$emit('console.log');
+        } ,     
+        onSubmit() {
+
+            if (this.editTransactionId) {
+                transaction = this.finalTransaction.find(t => t.id === this.editTransactionId);
+                console.log(transaction.id, transaction.name);
+                transaction.name = this.name;
+                transaction.description = this.description;
+                transaction.type = this.type;
+                transaction.amount = this.amount;
+                this.$emit("transaction-edited");
+                this.$emit('update-income');
+                this.$emit('update-expense');
+                this.$emit('update-currency');
+                this.name = null;
+                this.description = null;
+                this.type = null;
+                this.amount = null;
+            }
+            else {
+                let transactionList = {
```

Comment:
This is not a list. This is a new `transaction`.

Reply:
done